promql/parser/parse.go

```diff
@@ -443,6 +467,146 @@ func (p *parser) newAggregateExpr(op Item, modifier, args Node) (ret *AggregateE
 	return ret
 }
 
+// newMap is used when building the FloatHistogram from a map.
+func (p *parser) newMap() (ret map[string]interface{}) {
+	return map[string]interface{}{}
+}
+
+// mergeMaps is used to combine maps as they're used to later build the Float histogram.
+// This will merge the right map into the left map.
+func (p *parser) mergeMaps(left, right *map[string]interface{}) (ret *map[string]interface{}) {
+	for key, value := range *right {
+		if _, ok := (*left)[key]; ok {
+			p.addParseErrf(PositionRange{}, "duplicate key \"%s\" in histogram", key)
+			continue
+		}
+		(*left)[key] = value
+	}
+	return left
+}
+
+func (p *parser) histogramsIncreaseSeries(base, inc *histogram.FloatHistogram, times uint64) ([]SequenceValue, error) {
+	return p.histogramsSeries(base, inc, times, func(a, b *histogram.FloatHistogram) (*histogram.FloatHistogram, error) {
+		return a.Add(b)
```

Comment:
To avoid the error (or panic), you can do the following to make the addition always work:
```suggestion
	return p.histogramsSeries(base, inc, times, func(a, b *histogram.FloatHistogram) (*histogram.FloatHistogram, error) {
		if b.Schema >= a.Schema {
			return a.Add(b)
		}
		return b.Add(a)
```

Or if you _want_ an error, do the following:

```suggestion
	return p.histogramsSeries(base, inc, times, func(a, b *histogram.FloatHistogram) (*histogram.FloatHistogram, error) {
		if b.Schema < a.Schema {
			return nil, fmt.Errorf("some meaningful error message")
		}
		return a.Add(b)
```